query/tables/nutrient.py

```diff
@@ -0,0 +1,37 @@
+"""Full list of nutrient tags that appear on actual products"""
+
+from ..database import create_record, get_rows_affected
+
+
+async def create_table(transaction):
+    await transaction.execute(
+        'create table "nutrient" ("id" serial primary key, "tag" text not null);',
+    )
+    await transaction.execute(
+        'alter table "nutrient" add constraint "nutrient_tag_unique" unique ("tag");',
+    )
+
+
+async def create_nutrients_from_staging(transaction):
```

Comment:
Just curious: how big is the nutrient table? Since we let users add nutrients, there could be quite a few that are almost not used at all. Does it have a positive performance impact if we limit the nutrients to the ones in the nutrients taxonomy?

Reply:
With the 3,000 ish product sample data I was only getting 72 nutrients. I'm running locally now with a full MongoDB dump from a while ago to see how it looks.

My longer term goal here is to use this structure for other tags too to decrease the size of the main product_{tag} table which should improve query performance as per previous discussions with @cquest as the table will then just contain the product_id, tag_id (both ints) and obsolete flag.

We could then also load taxonomy data into these lookup tables to easily see where products are using values not in the taxonomy.

Reply:
Hi @stephanegigandet , on a full import I am getting 2237 nutrients so not too bad. I don't think that will have an adverse impact on performance.

Reply:
Analysis attached if you're interested.
```
select tag,count(*) products,min(value) minimum,avg(value) average,max(value) maximum from nutrient join product_nutrient on nutrient_id = id group by tag order by 2 desc;
```
[nutrient_tags.csv](https://github.com/user-attachments/files/20838601/nutrient_tags.csv)